### PR TITLE
[F2F-734] Updates to reflect IPV_F2F_CRI_VC_CONSUMED event sending nameParts

### PR DESCRIPTION
**Pull request by Caitlin-cooling**
## Proposed changes
As agreed [here](https://gds.slack.com/archives/C04N9K8GZDX/p1685821801352049?thread_ts=1685707211.898849&cid=C04N9K8GZDX) `nameParts` are now going to be sent in the `IPV_F2F_CRI_VC_CONSUMED` event. Proposed changes as detailed in the ticket include the update of `PostEventProcessor`  and `SessionEventProcessor` to handle the `nameParts` and include them in the call to govNotify.

Event structure to be is 
```
{
    "event_name": "IPV_F2F_CRI_VC_CONSUMED",
    "component_id": "https://g1vp8kzas9.execute-api.eu-west-2.amazonaws.com",
    "user": {
        "user_id": "urn:uuid:f81d4fae-7dec-11d0-a765-00a0c91e6bf6",
        "session_id": null,
        "govuk_signin_journey_id": "unknown",
        "ip_address": null
    },
    "restricted": {
        "nameParts": [
            {
                "type": "GivenName",
                "value": "ANGELA"
            },
            {
                "type": "GivenName",
                "value": "ZOE"
            },
            {
                "type": "FamilyName",
                "value": "UK SPECIMEN"
            }
        ]
    },
    "timestamp": 1685821531
}
```

### What changed
- `PostEventProcessor` updated to write the `nameParts` in to the db when `IPV_F2F_CRI_VC_CONSUMED` event is triggered
- `SessionEventProcessor` updated to send the `nameParts` to govNotify 
- `SessionEvent` type updated to include `nameParts`
- `PersonalIdentityUtils.getName` copied over from f2f to extract first and last name from the `nameParts` array

### Why did it change
We need to save `nameParts` from the `IPV_F2F_CRI_VC_CONSUMED` event in the db so that it can later be extracted and the users name can be included in the email to them.

### Screenshots

Event with updated payload structure sent
Screenshot 2023-06-19 at 11 36 20 am

NameParts showing on item in DDB
Screenshot 2023-06-19 at 11 36 55 am

### PII logging

- [x] Verified that no PII data is being logged

### Environment variables or secrets

- [x] No environment variables or secrets were added or changed

### CODE REVIEW

File: src/tests/unit/services/SessionEventProcessor.test.ts [hunk 1]
```diff
@@ -33,6 +33,14 @@ describe("SessionEventProcessor", () => {
 		const response = await sessionEventProcessorTest.processRequest(sessionEvent);
 		// eslint-disable-next-line @typescript-eslint/unbound-method
 		expect(mockIprService.sendToGovNotify).toHaveBeenCalledTimes(1);
+		expect(mockIprService.sendToGovNotify).toHaveBeenCalledWith({
```
Comment (Caitlin-cooling) [inline src/tests/unit/services/SessionEventProcessor.test.ts:36]: Add test for scenario where nameParts is not given/correct

**Comment by Caitlin-cooling** [comment]
TODO dev test this against ACs and attach screenshots of logs once I have AWS access

File: src/tests/unit/services/SessionEventProcessor.test.ts (hunk 1 above)
Comment (Caitlin-cooling) [reply src/tests/unit/services/SessionEventProcessor.test.ts:36]: Done

**Comment by Caitlin-cooling** [comment]
@richa-misra now that the event contains PII when we log the record [here](https://github.com/alphagov/di-ipvreturn-api/blob/F2F-734/src/PostEventHandler.ts#L25) we are logging PII. 

How would you normally handle this? I can see that the log would be helpful for debugging so do we have a pattern for obfuscating the data maybe?

**Review by hemavana — CHANGES_REQUESTED** [review]

**Comment by richa-misra** [comment]
> @richa-misra now that the event contains PII when we log the record [here](https://github.com/alphagov/di-ipvreturn-api/blob/F2F-734/src/PostEventHandler.ts#L25) we are logging PII.
> 
> How would you normally handle this? I can see that the log would be helpful for debugging so do we have a pattern for obfuscating the data maybe?

Hi @Caitlin-cooling good ques, yes we would need to implement a hashing pattern for PII fields across the logs. Instead of logging the entire SQS record, could you try logging the PII fields by replacing their actual value with its hash instead when logging it pls, something like this 

`****** (${Md5.hashStr(value.toString())})`

File: src/models/SessionReturnRecord.ts [hunk 2]
```diff
@@ -47,7 +48,7 @@ export class SessionReturnRecord {
 
     userEmail?: string;
 
-    nameParts?: PersonName[];
+    nameParts?: NamePart[];
```
[the remark this answers is not in the document]
Comment (Caitlin-cooling) [reply src/models/SessionReturnRecord.ts:48]: Done!

**Pull request merged**